compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIETileAndFuse.cpp

```diff
@@ -110,10 +125,21 @@ void AMDAIETileAndFusePass::runOnOperation() {
   func::FuncOp funcOp = getOperation();
 
   TilingInterface consumerOp;
+  TilingInterface matmulOp;
+  int64_t copyCnt = 0;
   funcOp->walk<WalkOrder::PostOrder, ReverseIterator>([&](TilingInterface op) {
+    // When there are copies of matmul operands and the targetOp is linalg.copy,
+    // different tiling strategies are applied on these linalg.copy. In the post
+    // order walk, linalg.copy ops are found in the order of (result, rhs, lhs).
+    // We only want to apply tiling on lhs and rhs of them in the pad-pack
+    // pipeline.
+    if (isa<linalg::CopyOp>(op)) copyCnt++;
+    if (isa<linalg::MatmulOp>(op)) matmulOp = op;
+
     // Find the next consumer op if it does not have loops OR it is from
-    // the skip ops list which currently contains linalg.copy and tensor.unpack.
-    if (op.getLoopIteratorTypes().empty() || consumerToSkip(op))
+    // the skip ops list.
+    if (op.getLoopIteratorTypes().empty() ||
+        consumerToSkip(op, targetOp, copyCnt))
```

Comment:
I am a bit concerned about the generality of the logic we're adding here. The logic here and above in `consumerToSkip` feels very specific to the `copy + matmul` pattern we see here during tiling?  Is there a way to generalize this more?

Reply:
I agree this is customized to the current IR, but I didn't find a better way to generalize this.

Reply:
> Confused by why it is a consumer op?

Normally the consumer is the tiling target e.g., matmulOp. I think it is set as "consumer" is because we want to fuse greedily to include its producers.